x-pack/plugins/monitoring/public/components/cluster/listing/listing.js

```diff
@@ -333,124 +331,123 @@ const handleClickInvalidLicense = (scope, clusterName) => {
+      { theme$ }
     ),
   });
 };
 
-export class Listing extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      [STANDALONE_CLUSTER_STORAGE_KEY]: false,
-    };
+const StandaloneClusterCallout = ({ changeCluster, storage }) => {
+  if (storage.get(STANDALONE_CLUSTER_STORAGE_KEY)) {
+    return null;
   }
 
-  renderStandaloneClusterCallout(changeCluster, storage) {
-    if (storage.get(STANDALONE_CLUSTER_STORAGE_KEY)) {
-      return null;
-    }
-
-    return (
-      <div>
-        <EuiCallOut
-          color="warning"
-          title={i18n.translate('xpack.monitoring.cluster.listing.standaloneClusterCallOutTitle', {
-            defaultMessage:
-              "It looks like you have instances that aren't connected to an Elasticsearch cluster.",
-          })}
-          iconType="link"
-        >
-          <p>
-            <EuiLink
-              onClick={() => changeCluster(STANDALONE_CLUSTER_CLUSTER_UUID)}
-              data-test-subj="standaloneClusterLink"
-            >
-              <FormattedMessage
-                id="xpack.monitoring.cluster.listing.standaloneClusterCallOutLink"
-                defaultMessage="View these instances."
-              />
-            </EuiLink>
+  return (
+    <div>
+      <EuiCallOut
+        color="warning"
+        title={i18n.translate('xpack.monitoring.cluster.listing.standaloneClusterCallOutTitle', {
+          defaultMessage:
+            "It looks like you have instances that aren't connected to an Elasticsearch cluster.",
+        })}
+        iconType="link"
+      >
+        <p>
+          <EuiLink
+            onClick={() => changeCluster(STANDALONE_CLUSTER_CLUSTER_UUID)}
+            data-test-subj="standaloneClusterLink"
+          >
+            <FormattedMessage
+              id="xpack.monitoring.cluster.listing.standaloneClusterCallOutLink"
+              defaultMessage="View these instances."
+            />
+          </EuiLink>
+          &nbsp;
+          <FormattedMessage
+            id="xpack.monitoring.cluster.listing.standaloneClusterCallOutText"
+            defaultMessage="Or, click Standalone Cluster in the table below"
+          />
+        </p>
+        <p>
+          <EuiLink
+            onClick={() => {
+              storage.set(STANDALONE_CLUSTER_STORAGE_KEY, true);
+            }}
+          >
+            <EuiIcon type="cross" />
             &nbsp;
             <FormattedMessage
-              id="xpack.monitoring.cluster.listing.standaloneClusterCallOutText"
-              defaultMessage="Or, click Standalone Cluster in the table below"
+              id="xpack.monitoring.cluster.listing.standaloneClusterCallOutDismiss"
+              defaultMessage="Dismiss"
             />
-          </p>
-          <p>
-            <EuiLink
-              onClick={() => {
-                storage.set(STANDALONE_CLUSTER_STORAGE_KEY, true);
-                this.setState({ [STANDALONE_CLUSTER_STORAGE_KEY]: true });
```

Comment:
Noting that I've intentionally removed this setState since it doesn't appear to get read anywhere.

Reply:
It seems to be set on line 384 in order to force a re-render to return null on line 351.

Reply:
good catch, we hoped that wouldn't be the case when converting the component :grimacing: looks like we need a

```typescript
const [_, forceRerender] = useState({});
```

Reply:
Nice catch! I opened up https://github.com/elastic/kibana/issues/121605 to address